proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -298,6 +297,9 @@ public ProctorResult determineTestGroups(
             }
         }
 
+        final Map<String, ValueExpression> localContext =
```

Comment:
`localContext` was initialized inside `TestChooser`, so it's recalculated for each iteration.